Tools/TestWebKitAPI/Tests/JavaScriptCore/glib/TestJSC.cpp

```diff
@@ -164,23 +164,27 @@ static void testJSCBasic()
         checker.watch(context.get());
         ExceptionHandler exceptionHandler(context.get());
 
+        bool tag;
```

Comment:
Heh, smart! Good thinking.

I would simplify this though:

 * You don't actually need the tag variable. This is a little confusing because if you don't squint hard enough, you might think that the code is using uninitialized memory by mistake. I see that what you've written is _correct_ -- you just need the address, and never access the value -- but it looks suspicious if you don't look closely enough. You don't actually need to store a pointer value; storing something like `GINT_TO_POINTER (TRUE)` would suffice.
 * I would either name it "is-original-jscvalue" or else add a short comment to explain why the tag exists. Probably just adjusting the name would be sufficient for people reading the code to immediately understand what you're doing here.

So you might have, say:

```
g_object_set_data(G_OBJECT(result.get()), "is-original-jscvalue", GINT_TO_POINTER(TRUE));
```

And:

```
g_assert_cmpint(GPOINTER_TO_INT(g_object_get_data(G_OBJECT(result.get()), "is-original-jscvalue"), ==, TRUE);
```

Although, hm, maybe the GPOINTER macros are a little unwieldy, which doesn't help readability. So, you get to choose whether you stick with your stack tag or choose to store just `TRUE` instead. I'm going to approve this either way.

Reply:
Thank you for your suggestion. I like it.

How about this one?
```
g_assert_true(GPOINTER_TO_INT(g_object_get_data(G_OBJECT(result.get()), "is-original-jscvalue")))
```
It's a little bit shorter.